frontend/src/employee-frontend/components/person-shared/PersonDetails.tsx

```diff
@@ -127,6 +128,8 @@ export default React.memo(function PersonDetails({
     forceManualFeeDecisions: false,
     ophPersonOid: ''
   })
+  const [emailInfo, setEmailInfo] = useState<InputInfo | undefined>(undefined)
+  const [emailIsValid, setEmailIsValid] = useState<boolean>(true)
```

Comment:
Näitä on turha pyörittää staten kautta koska ne voidaan laskea suoraan muista arvoista. Nyt tulee turhaa koodia settereiden kutsumisesta ja altistaa bugeille kun voi mennä epäsynkkaan. Nää on molemmat niin yksinkertaisia ettei periaatteessa haittaa laskea uudelleen vaikka joka renderillä, mutta mallin vuoksi tuon validointituloksen laskemisen voi optimoida näin
```
const emailIsValid = useMemo(() =>
   form.email === form.email.trim(),
   [form.email]
)
```
ja sitten tuon infon voi laittaa suoraan komponentin propsiin
```
info={emailIsValid ? undefined : {
   text: i18n.validationErrors.email,
   status: 'warning'
}}
```